src/components/menu/MenuGroup.tsx

```diff
@@ -0,0 +1,15 @@
+import { getValidMenuChildren } from "./menuUtils";
+
+const MenuGroup = ({
+  children,
+  className = "",
+}: {
+  children: React.ReactNode;
+  className?: string;
+}) => {
+  const menuChildren = getValidMenuChildren(children, ["MenuGroup"]);
```

Comment:
what's the reason for filtering out MenuGroups? At best it'll slow things down, and at worst it'll prevent nested MenuGroups.

That said, right now the reason for `<MenuGroup>` compo is only so that apps can render a group of items since we filter every non-whitelisted nodes out in `getValidMenuChildren()` (that "arbitrary" gap on the group should be removed, and added in host).

I'd retain the filtering (and thus `MenuGroup` compo) only if we decide to have a hybrid solution where we render default items together with custom ones — in which case having named components will be necessary (almost). Otherwise can let hosts render whatever they want.

Reply:
I am not sure which `gap` you are talking about, I mistakenly left some scss in styles.scss which is now moved to excal app so probably that style you are talking about

As per `MenuGroup` we should definitely keep it, the reason being its a wrapper for grouping a set of items (be it custom or default) so that the items could be grouped together and styled easily as well and whatever default styles get applied to group will be applied as well (rn we don't have but we can have in future), adding header to groups could be easier as well. And later when we make it more accessible it will be supported out of the box. 
And allowing to render everything is not a good idea as it will have downsides - achieving consistent design, making it accessible etc hence I want to allow only named items and not anything inside `MenuGroup`. 
Overall the API becomes easier to use for the host and has benefits as mentioned above.

As per `render` attribute I didn't really get what the usage of `render` attribute, I would completely leave it up to host how they want to render the items if they are rendering the menu on their own so they render only what they want and reorder and have full control as well.